include/up-cpp/datamodel/validator/UMessage.h

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-FileType: SOURCE
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_CPP_DATAMODEL_VALIDATOR_UMESSAGE_H
+#define UP_CPP_DATAMODEL_VALIDATOR_UMESSAGE_H
+
+#include <uprotocol/v1/umessage.pb.h>
+
+/// @brief Validators for UMessage objects.
+///
+/// See
+/// https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uattributes.adoc
+namespace uprotocol::datamodel::validator::message {
+
+/// @brief Checks if UMessage is a valid UMessage of any format.
+///
+/// A UMessage is valid if any of these are true:
+///
+///   * isValidRpcRequest()
+///   * isValidRpcResponse()
+///   * isValidPublish()
+///   * isValidNotification()
+[[nodiscard]] bool isValid(const v1::UMessage&);
```

Comment:
What's the use case for this?

Reply:
`UTransport::send()`

Reply:
Ahhh makes sense. We don't care in that case what it is, just that it's _some_ type of valid UMessage.